pvlib/tests/test_pvarray.py

```diff
@@ -69,3 +69,45 @@ def test_huld():
     with pytest.raises(ValueError,
                        match='Either k or cell_type must be specified'):
         res = pvarray.huld(1000, 25, 100)
+
+
+@pytest.mark.parametrize('method', ['ols', 'robust'])
+@requires_statsmodels
+def test_fit_huld(method):
+    # test is to recover the parameters in _infer_huld_k for each cell type
+    # IEC61853 conditions to make data for fitting
+    ee, tc = pvarray._build_iec61853()
+    techs = ['csi', 'cis', 'cdte']
+    pdc0 = 250
+    for tech in techs:
+        k0 = pvarray._infer_k_huld(tech, pdc0)
+        pdc = pvarray.huld(ee, tc, pdc0, cell_type=tech)
+        m_pdc0, k = pvarray.fit_huld(ee, tc, pdc, method=method)
+        expected = np.array([pdc0, ] + [v for v in k0], dtype=float)
+        modeled = np.hstack((m_pdc0, k))
+        assert_allclose(expected, modeled, rtol=1e-8)
+    # once more to check that NaNs are handled
+    ee[7] = np.nan
+    tc[9] = np.nan
+    k0 = pvarray._infer_k_huld('csi', pdc0)
+    pdc = pvarray.huld(ee, tc, pdc0, cell_type='csi')
+    pdc[11] = np.nan
+    m_pdc0, k = pvarray.fit_huld(ee, tc, pdc, method='ols')
+    # known solution for OLS obtained with statsmodels v0.14.0
+    expected = np.array([2.07118648e+02, -8.35033390e+01, -3.83844606e+01,
+                         2.75629738e+00,  1.87571414e+00, -2.86429730e-01,
+                         -6.00418853e-02])
```

Comment:
I don't understand why the coefficients are expected to be different in the presence of NaN.  The Huld model perfectly describes the remaining data, and there are enough points remaining to uniquely specify the coefficients, so why shouldn't the source coefficients `k0` be returned exactly?

Filtering the NaNs manually causes the source coefficients to be successfully recovered:

```python
ee, tc = _build_iec61853()
pdc0 = 250
ee[7] = np.nan
tc[9] = np.nan
k0 = _infer_k_huld('csi', pdc0)
pdc = huld(ee, tc, pdc0, cell_type='csi')
pdc[11] = np.nan

# returns source coefficients
filt = ~(np.isnan(ee) | np.isnan(tc) | np.isnan(pdc))
m_pdc0, k = fit_huld(ee[filt], tc[filt], pdc[filt], method='ols')

# returns different coefficients, why?
m_pdc0, k = fit_huld(ee, tc, pdc, method='ols')
```

Reply:
Aha. The introduced NaN in `ee` is being overwritten in `fit_huld` with a 0.